Language/Statements/Local_Variable_Declaration/reference_error_A01_t02.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023 the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if a local variable is referenced at a
+/// source code location that is before the end of its initializing expression,
+/// if any, and otherwise before the declaring occurrence of the identifier
+/// which names the variable.
+///
+/// @description Checks that a variable is introduced into the scope after a
+/// variable declaration statement is evaluated, and the name of this variable
```

Comment:
That isn't quite accurate: The local declaration introduces the name into the enclosing scope from the beginning of that scope:

```dart
var x = 1;

void main() {
  print(x); // Error (`x` is the local variable below, not the top-level variable).
  var x = true;
}
```

But it is true that we can't use that name (it's a compile-time error no matter how we use it) before the end of the initializing expression in the declaration of that name.